lib/active_admin/middleware/error_handler.rb

```diff
@@ -0,0 +1,51 @@
+module ActiveAdmin
+  class ErrorHandler
+    include MethodOrProcHelper
+
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      @app.call(env)
+    rescue Exception => exception
+      if is_active_admin_error?(env) && use_active_admin_error_page?
+        render_exception(exception, env)
+      else
+        raise exception
+      end
+    end
+
+    private
+
+    def is_active_admin_error?(env)
+      stringified_namespaces.include?(current_namespace(env))
+    end
+
+    def use_active_admin_error_page?
+      render_in_context(ActiveAdmin.application, ActiveAdmin.application.use_active_admin_error_page)
```

Comment:
`render_in_context(self, ActiveAdmin.application.use_active_admin_error_page)` should work